### PR TITLE
css-layout-classes - Lawrence Wang

### PR DESCRIPTION
**Pull request by wang-lawrence**
Screen Shot 2023-05-17 at 10 53 19 PM

I have a couple of questions on this exercise. 
1. Without setting the image to 100% width, the row with 4 images overflowed. I thought setting the flex-basis = 25% would shrink the images so they fit into the container, why didn't it? Should we always set images to width to 100% just to be safe?
Screen Shot 2023-05-17 at 9 37 42 PM
2. I shrunk the images on the row with four images. I then tried to center them by selecting the column-fourth class (the div's that contain each image) and setting justify-content: center, but the images didn't move. How would I go about centering these images?

Screen Shot 2023-05-17 at 10 03 57 PM

### CODE REVIEW

File: css-layout-classes/index.html [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>CSS Layout Classes</title>
+  <link rel="stylesheet" href="styles.css">
+</head>
+<body>
+  <div class="container">
+    <h1>Choose your character!</h1>
```
Comment (shawnkost) [inline css-layout-classes/index.html:12]: This should be inside a `row` & a `column-full`

File: css-layout-classes/index.html [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    <div class="row">
+      <div class="link-image">
```
Comment (shawnkost) [inline css-layout-classes/index.html:14]: This should be named more appropriately like `column-third`

File: css-layout-classes/index.html [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        <img src="images/link.png" alt="botw link in champions tunic">
+      </div>
+      <div class="ssb-text">
```
Comment (shawnkost) [inline css-layout-classes/index.html:17]: This should be named more appropriately like `column-third`

**Comment by wang-lawrence** [comment]
You've actually answered all my questions in class so we're good!

File: css-layout-classes/index.html [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        <h2>Super Smash Bros. Ultimate</h2>
+        <p>Super Smash Bros. Ultimate received universal acclaim from both critics and players, with some critics calling it the best installment in the series.
+          It received praise for its large amount of content and fine-tuning of existing Smash gameplay elements, although its online mode was widely criticized.
+        </p>
+      </div>
+      <div class="yoshi-image">
```
Comment (shawnkost) [inline css-layout-classes/index.html:23]: This should be named more appropriately like `column-third`

File: css-layout-classes/index.html [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      </div>
+      <div class="column-fourth">
+        <img src="images/pikachu.png" alt="cute pikachu" class="character">
+      </div>
+      <div class="column-fourth">
+        <img src="images/ness.png" alt="ness posing" class="character">
+      </div>
+    </div>
+    <div class="row">
+      <div class="dk-text">
```
Comment (shawnkost) [inline css-layout-classes/index.html:42]: This should be named more appropriately like `column-two-thirds`

File: css-layout-classes/index.html [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        <h2>Donkey Kong</h2>
+        <p>Donkey King is a series of video games featuring the adventures of a gorilla called Donkey King, conveived by Shiegru Myamoto in 1981.
+          The franchise consists mainly of two game genres, but also includes additional spin-off titles of various genres.
+          <br>
+          <br>
+          The games of the first genre are mostly single-screen platform/action puzzle types, featuring Donkey King as the opponent in an industrial construction setting.
+        </p>
+      </div>
+      <div class="dk-image">
```
Comment (shawnkost) [inline css-layout-classes/index.html:51]: This should be named more appropriately like `column-third`

File: css-layout-classes/index.html [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        <img src="images/dk.png" alt="DK punching the air" >
+      </div>
+    </div>
+    <div class="row">
+      <div class="mario-image">
```
Comment (shawnkost) [inline css-layout-classes/index.html:56]: This should be named more appropriately like `column-third`

File: css-layout-classes/index.html [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        <img src="images/mario.png" alt="Mario fly kick">
+      </div>
+      <div class = "mario-text">
```
Comment (shawnkost) [inline css-layout-classes/index.html:59]: This should be named more appropriately like `column-two-thirds`

File: css-layout-classes/styles.css [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+.row {
+  display: flex;
+}
+
+.column-fourth {
+  flex-basis: calc(100% /4);
+}
+
+
```
Comment (shawnkost) [inline css-layout-classes/styles.css:18]: Once you adjust your class names, you'll need to adjust the css. You should have more column selectors here. Compare against the solution I posted in the main class channel

**Review by shawnkost — CHANGES_REQUESTED** [review]
Some changes requested here Lawrence!

**Review by shawnkost — APPROVED** [review]
👍🏻

**Pull request merged**